scripts/pylib/twister/twisterlib/runner.py

```diff
@@ -784,7 +784,7 @@ def cmake(self):
 
         # merge overlay files into one variable
```

Comment:
I believe it is worth to explain the modified mechanism in a comment. E.g. expand the comment with: "Overlays with prefixes won't be merged but pass to cmake as they are.

Reply:
Will this commit message be ok:  
This commit changes regular expression used to merge overlays 
specified for test samples in twister tests.

Overlays without any prefixes will be merged and passed to cmake.
Overlays with prefixes won't be merged but passed to cmake as they are.

Reply:
What I wish to point out - that in the form before changes, all overlays with prefixes was merged but to the one variable without prefix. So the mechanism was broken by design, and this commit fixes it, but I believe we cannot use word "fix" in a commit.

Reply:
Yes - it was broken. But per my understanding child images are not use in the upstream so it was never an issue here. Could You please accept the changes if it is OK now.